tests/marray/marray_constructor.h

```diff
@@ -31,81 +31,125 @@ template <typename DataT, typename NumElementsT>
 class run_marray_constructor_test {
   static constexpr std::size_t NumElements = NumElementsT::value;
 
-  using marray_t = sycl::marray<DataT, NumElements>;
+  static constexpr size_t num_test_cases = 7;
 
- private:
-  template <size_t num_elements = NumElements,
-            std::enable_if_t<num_elements == 1, bool> = true>
-  void check_constexpr_single_element() {}
+  static constexpr const char* check_names[num_test_cases] = {
+      "default constructor",
+      "scalar constructor",
+      "variadic constructor (NumElements DataT instances)",
+      "variadic constructor (one DataT instance, one marray instance)",
+      "variadic constructor (one marray instance, one DataT instance)",
+      "copy constructor",
+      "copy constructor rval reference"};
 
-  template <size_t num_elements = NumElements,
-            std::enable_if_t<num_elements != 1, bool> = true>
-  void check_constexpr_single_element() {
-    //  one DataT instance, one marray instance
-    {
-      constexpr sycl::marray<DataT, num_elements - 1> ma_const =
-          marray_common::iota_marray<DataT, num_elements - 1, 2>();
-      constexpr marray_t ma{1, ma_const};
-      marray_t ma_inc;
-      std::iota(ma_inc.begin(), ma_inc.end(), 1);
-      CHECK(value_operations::are_equal(ma_inc, ma));
-    }
-
-    // one marray instance, one DataT instance
-    {
-      constexpr sycl::marray<DataT, num_elements - 1> ma_const =
-          marray_common::iota_marray<DataT, num_elements - 1, 1>();
-      constexpr marray_t ma{ma_const, DataT(num_elements)};
-      marray_t ma_inc;
-      std::iota(ma_inc.begin(), ma_inc.end(), 1);
-      CHECK(value_operations::are_equal(ma_inc, ma));
-    }
-  }
-
- public:
-  void operator()(const std::string&) {
-    INFO("for number of elements \"" << NumElements << "\": ");
+  using marray_t = sycl::marray<DataT, NumElements>;
 
+  template <typename IteratorT>
+  static void run_checks(IteratorT results) {
     // default constructor
     {
       marray_t ma;
-      CHECK(value_operations::are_equal(ma, DataT{}));
+      *(results++) = value_operations::are_equal(ma, DataT{});
     }
 
     // scalar constructor
     {
       constexpr DataT value{1};
       constexpr marray_t ma{value};
-      CHECK(value_operations::are_equal(ma, value));
+      *(results++) = value_operations::are_equal(ma, value);
     }
 
     // variadic constructor
     {
       // NumElements DataT instances
       constexpr auto a = marray_common::iota_marray<DataT, NumElements, 1>();
       marray_t ma_inc;
-      std::iota(ma_inc.begin(), ma_inc.end(), 1);
-      CHECK(value_operations::are_equal(ma_inc, a));
+      marray_common::iota(ma_inc.begin(), ma_inc.end(), 1);
+      *(results++) = value_operations::are_equal(ma_inc, a);
 
       // only compiled when NumElements != 1
-      check_constexpr_single_element();
+      if constexpr (NumElements != 1) {
```

Comment:
What is the problem with 1?

Reply:
These test cases try to construct an `marray` using a scalar and an `marray` of size `NumElements - 1`. If `NumElements` is 1, then the smaller `marray` would have to be of size 0. 0-sized `marray`s are not allowed by the [specification](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:marray.type) based on the following:
> The number of elements parameter, NumElements, is a positive value of the std::size_t type.

Reply:
Oh I see, by looking at the called function.
You just implement the comment, actually. :-)